### PR TITLE
Setup mailer & email verification

### PR DESCRIPTION
**Pull request by s-kakol**
* `.env.example` - Added new env variables, modified the example file to update.
* `package-lock.json`, `package.json` - Add nodemailer + types dependencies.
* `src/auth/auth.controller.ts` - 
  1. Register Endpoint:  If registration ends successfully send verification email.
  2. Verification Endpoint: `@ApiExcludeEndpoint()` - hides endpoint from swagger docs as it's for internal use only. `@Public()` because it doesn't need JWT token.
* `src/auth/auth.module.ts` - Add EmailModule/Service to Auth Module.
* `src/auth/auth.service.ts` - Couple of methods:
  1. `sendVerificationLink()` - Creates JWT token to be sent as a query param, calls EmailService to send an email.
  2. `confirmEmailVerification()` - First it verifies the JWT token, checks if email is in the payload and continues to double check if user has already verified email. If yes throws error, if he didn't calls UserService `updateUserEmailVerification()`.
* `src/common/utilities/config-validation.schema.ts` - Joi validation expanded to accommodate new env variables.
* `src/email/email.module.ts` - Email module setup, basic. Also has ConfigModule as it refers to env variables.
* `src/email/email.service.ts` - Basic construction, helper method to send emails.
* `src/user/models/user.entity.ts` - Add `has_verified_email` column to User Entity.
* `src/user/user.service.ts` - During user creation set `has_verified_email` as default false. Simple method `updateUserEmailVerification` to find user by ID and set `has_verified_email` to true.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -21,3 +21,21 @@ NEST_PORT
 JWT_SECRET
 # Expects a (value-name) string, eg.: '1w', '3600s'
 JWT_EXPIRE
+
+# Expects a valid email service, eg.: 'gmail'
+EMAIL_SERVICE
```
Comment (adjurk) [inline .env.example:26]: What are other possible options? SMTP? Could you provide the package docs for this value?

Comment (adjurk) [reply .env.example:26]: Also, can you set an SMTP port and host then?

**Review by adjurk — APPROVED** [review]
I'd appreciate supporting SMTP if possible, since Gmail may not be a reliable solution for production use. Other than that, it's _good enough_ at this point.

File: .env.example (hunk 1 above)
Comment (s-kakol) [reply .env.example:26]: https://community.nodemailer.com/2-0-0-beta/setup-smtp/well-known-services/

Possibly, didn't try that.

**Pull request merged**